gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjureBasePlugin.java

```diff
@@ -98,6 +116,9 @@ private static TaskProvider<CompileIrTask> createIrTasks(
             compileIr.getExecutableDir().set(extractCompilerTask.flatMap(ExtractExecutableTask::getOutputDirectory));
             compileIr.getOutputIrFile().set(irDir.map(dir -> dir.file(project.getName() + ".conjure.json")));
             compileIr.getProductDependencies().set(project.provider(pdepsExtension::getProductDependencies));
+            compileIr
+                    .getOptions()
+                    .set(project.provider(() -> conjureExtension.getParser().getProperties()));
```

Comment:
Afaik, general best practice is to make all fields of an extension (i.e. `conjureExtension`) a property and that way you get the lazy wiring out of the box. But this works as well and don't think we need to refactor the extension now.

Reply:
Yeah, I would just leave it like this! Just wanted to flag but don't think its a big deal!

Reply:
It definitely is nicer to just hook up the properties together directly, especially as Gradle can then work out the task dependencies automatically if there are files involved in outputs and inputs of different tasks. The one thing I will say is the extra providers makes the stacktrace more understandable when you have some horrible not-enough-laziness bug like I recently encountered with gradle-jdks, but I don't think that's a good enough reason to not just link them together (and you can always add these back when debugging).

Reply:
I've filed #1050 to track